enzy_htp/_interface/amber_interface.py

```diff
@@ -1485,16 +1519,28 @@ def _parse_md_config_dict_to_raw(self, md_config_dict: Dict) -> Dict:
             geom_cons: List[StructureConstraint]
             nmropt_cntrl = {'nmropt': 1}
             # figure out path
-            disang_path: str = geom_cons[0].params["amber"]["rs_filepath"]
+            if geom_cons[0].is_residue_pair_constraint():
+                for (_, cst) in geom_cons[0].child_constraints:
+                    disang_path: str = cst.params["amber"]["rs_filepath"]
+                    break
+                else:
+                    assert False
+                    #TODO(CJ): figure this out
+            else:
+                disang_path: str = geom_cons[0].params["amber"]["rs_filepath"]
             if disang_path.startswith("{mdstep_dir}"):
                 disang_path = disang_path.lstrip("{mdstep_dir}")
                 disang_path = f"{md_config_dict['mdstep_dir']}/{disang_path}"
             disang_path = fs.get_valid_temp_name(disang_path)
             # figure out content
             disang_content_list = []
             for cons in geom_cons:
-                raw_rs_dict = self._parse_cons_to_raw_rs_dict(cons)
-                disang_content_list.append(raw_rs_dict)
+                if cons.is_residue_pair_constraint():
+                    for (_, cst) in cons.child_constraints:
+                        disang_content_list.append( self._parse_cons_to_raw_rs_dict(cst) )
+                else:                    
+                    raw_rs_dict = self._parse_cons_to_raw_rs_dict(cons)
+                    disang_content_list.append(raw_rs_dict)
```

Comment:
I guess a good treatment would be if residue pair constraint is found, extend the geom_cons list by .child_constraints

Reply:
The current way is pretty hard to read and repeatative

Reply:
Yeah I think so too. I just left it that way to get it to work for an example. I will fix it

Reply:
reverted to previous with commit 1